### PR TITLE
compiler/natives/src/reflect: fix valueIntrface check struct copy

### PR DESCRIPTION
**Pull request by visualfc**
fix https://github.com/gopherjs/gopherjs/issues/1156

### CODE REVIEW

**Comment by nevkontakte** [comment]
A couple of questions:

 - Should the copy happen when `Value.Index()` or `Value.Interface()` method is called? I think in the equivalent non-reflect code the copy would happen on the index operation.
 - Does `reflectlite` have the same bug? If so, maybe fix it as well?

**Comment by visualfc** [comment]
> A couple of questions:
> 
> * Should the copy happen when `Value.Index()` or `Value.Interface()` method is called? I think in the equivalent non-reflect code the copy would happen on the index operation.

copy struct on reflect.Value to interface time.
  v.Index(0) == a[0]
```
	a := []student{student{"A"}, student{"B"}, student{"C"}}
	v := reflect.ValueOf(a)
	i := v.Index(0)
        j := i.Interface()
	a[0] = student{"X"}
	fmt.Println(a, i, j)
        // output [{X} {B} {C}] {X} {A}
```

> * Does `reflectlite` have the same bug? If so, maybe fix it as well?

sync with reflect

File: tests/copy_test.go [hunk 1]
```diff
@@ -131,3 +132,16 @@ func TestExplicitConversion(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestCopyStructByReflect(t *testing.T) {
+	type Info struct {
```
Comment (nevkontakte) [inline tests/copy_test.go:137]:
```suggestion
func TestCopyStructByReflect(t *testing.T) {
	// https://github.com/gopherjs/gopherjs/issues/1156
	type Info struct {
```

For such corner cases I usually add a reference to the original issue in the test case, this helps with code archeology a year or two later :)

**Review by nevkontakte — APPROVED** [review]

**Comment by nevkontakte** [comment]
Thank you for your contribution!
